### PR TITLE
Fix config

### PR DESCRIPTION
**Pull request by b1ron**
Refs #556, #557.

### CODE REVIEW

File: tests/mongo-go-driver.yml [hunk 1]
```diff
@@ -87,8 +87,8 @@ results:
   ferretdb:
     stats:
       expected_fail: 10
-      expected_skip: 19
-      expected_pass: 633
```
Comment (AlekSi) [inline tests/mongo-go-driver.yml:91]: Why the number of passing tests went down?

File: tests/mongo-go-driver.yml [hunk 2]
```diff
@@ -157,6 +155,8 @@ results:
       - go.mongodb.org/mongo-driver/x/mongo/driver/integration/TestAggregate/Multiple_Batches
       - go.mongodb.org/mongo-driver/x/mongo/driver/integration/TestAggregate/TestMaxTimeMSInGetMore
     fail:
+      - go.mongodb.org/mongo-driver/mongo/integration/TestCausalConsistency_NotSupported
```
Comment (AlekSi) [inline tests/mongo-go-driver.yml:158]: There must be a comment explaining why those tests fail

File: tests/mongo.yml [hunk 3]
```diff
@@ -181,3 +185,18 @@ results:
       - mongo/jstests/aggregation/use_query_project_and_sort.js
       - mongo/jstests/aggregation/use_query_projection.js
       - mongo/jstests/aggregation/use_query_sort.js
+      # uncaught exception: SyntaxError: import declarations may only appear at top level of a module
+      # all these syntax errors are fixed in MongoDB 7.0.
```
Comment (AlekSi) [inline tests/mongo.yml:189]: What does it mean? Isn't that PR updates mongo repo to 7.0?

Comment (b1ron) [reply tests/mongo.yml:189]: #559 wasn't merged so we were still using the older version of the [shell](https://github.com/FerretDB/dance/blob/main/build/deps/mongo.Dockerfile#L1).

I will approve #559 and then I can remove those failures.

File: tests/mongo-go-driver.yml (hunk 1 above)
Comment (b1ron) [reply tests/mongo-go-driver.yml:91]: I think it was caused by the `mongo` submodule being out of sync with shell dependency, #564 should fix this.

File: tests/mongo.yml (hunk 3 above)
Comment (AlekSi) [reply tests/mongo.yml:189]: (and https://github.com/FerretDB/deps/pull/115)

File: tests/mongo-go-driver.yml (hunk 1 above)
Comment (b1ron) [reply tests/mongo-go-driver.yml:91]: That's needs some investigation.

File: tests/mongo-go-driver.yml (hunk 2 above)
Comment (b1ron) [reply tests/mongo-go-driver.yml:159]: They were previously [passing](https://github.com/FerretDB/dance/blob/main/tests/mongo-go-driver.yml#L127-L128) now they panic.

```go.mongodb.org/mongo-driver/mongo/integration/TestCausalConsistency_NotSupported:
	=== RUN   TestCausalConsistency_NotSupported
	--- FAIL: TestCausalConsistency_NotSupported (0.08s)
	
go.mongodb.org/mongo-driver/mongo/integration/TestCausalConsistency_NotSupported/afterClusterTime_not_included:
	=== RUN   TestCausalConsistency_NotSupported/afterClusterTime_not_included
	    --- FAIL: TestCausalConsistency_NotSupported/afterClusterTime_not_included (0.08s)
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
		panic: runtime error: invalid memory address or nil pointer dereference
	[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe47b76]
	
	goroutine 72 [running]:
	testing.tRunner.func1.2({0xecf8e0, 0x17877e0})
		/usr/local/go/src/testing/testing.go:1526 +0x372
	testing.tRunner.func1()
		/usr/local/go/src/testing/testing.go:1529 +0x650
	panic({0xecf8e0, 0x17877e0})
		/usr/local/go/src/runtime/panic.go:890 +0x263
	go.mongodb.org/mongo-driver/mongo/integration.TestCausalConsistency_NotSupported.func1(0xc0002f0c40)
		/home/byron/workdir/dance/tests/mongo-go-driver/mongo/integration/causal_consistency_test.go:225 +0x256
	go.mongodb.org/mongo-driver/mongo/integration/mtest.(*T).RunOpts.func1(0x0?)
		/home/byron/workdir/dance/tests/mongo-go-driver/mongo/integration/mtest/mongotest.go:266 +0x5a9
	testing.tRunner(0xc0002fbba0, 0xc00003cde0)
		/usr/local/go/src/testing/testing.go:1576 +0x217
	created by testing.(*T).Run
		/usr/local/go/src/testing/testing.go:1629 +0x806
```

Comment (b1ron) [reply tests/mongo-go-driver.yml:159]: I am not super confident on this but it seems that the changes [made](https://github.com/mongodb/mongo-go-driver/compare/6d814184973ae7ffc13281c93b423dc522a4bcca...6c7e1241b747d2fb5d164c9ca9d17617f33e7899#diff-9d480eb84448ec1035a67cc3d37fd15d2aa1eb958ea6f8fb0d1c3b0d8ac85b27L226) now cause their event monitor to return nil [here](https://github.com/mongodb/mongo-go-driver/blob/67f257ac1d970452575fa24d7394e1ed46d1d725/mongo/integration/causal_consistency_test.go#L224) when it tries to inspect the command name. Could be related to https://github.com/mongodb/mongo-go-driver/pull/1347?

@AlekSi should I add this failing test to common ignore for now?

Comment (b1ron) [reply tests/mongo-go-driver.yml:159]: But it only fails on FerretDB which invalidates the above.

File: tests/mongo-go-driver.yml (hunk 1 above)
Comment (b1ron) [reply tests/mongo-go-driver.yml:91]: We seem to be running less subtests in total with the update. This is _not_ the case for MongoDB. I am unsure why this is happening and I took a lot at some of the [commits](https://github.com/mongodb/mongo-go-driver/compare/6d814184973ae7ffc13281c93b423dc522a4bcca...6c7e1241b747d2fb5d164c9ca9d17617f33e7899) for possible causes and could not find any. I logged some more output to verify that:

```
FerretDB:

CURRENT
commit: 6d814184973ae7ffc13281c93b423dc522a4bcca
Expectedly ignored: 16
Total number of tests: 678

UPDATE
commit: 6c7e1241b747d2fb5d164c9ca9d17617f33e7899
Expectedly ignored: 15
Total number of tests: 662
```

Any ideas?

**Pull request closed without merging**